File: mesi.io.domain/clipboard/src/main/kotlin/mesi/io/domain/clipboard/ClipboardService.kt
package mesi.io.domain.clipboard

import mesi.io.data.clipboard.ClipboardEntry

/**
 * Access to clipboard entries
 */
interface ClipboardService {

    /**
     * Returns a clipboard entry with given id
     */
    fun getById(id : String) : ClipboardEntry?

    /**
     * Returns all clipboard entries
     */
    fun getAll() : List<ClipboardEntry>

    /**
     * Adds a new clipboard entry with given content
     */
    fun add(content : String)
}

File: src/Mesi.Io.Api.Clipboard/Domain/ClipboardService.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using Mesi.Io.Api.Clipboard.Data;
using Mesi.Io.Api.Clipboard.Models;

namespace Mesi.Io.Api.Clipboard.Domain
{
    public class ClipboardService : IClipboardService
    {
        private readonly IClipboardRepository _clipboardRepository;

        public ClipboardService(IClipboardRepository clipboardRepository)
        {
            _clipboardRepository = clipboardRepository;
        }

        public void AddEntryForUser(Guid userId, string content)
        {
            _clipboardRepository.AddEntry(new ClipboardEntry(userId, content, DateTime.UtcNow));
        }

        public Task<IEnumerable<ClipboardEntry>> GetAllByUserId(Guid userId)
        {
            return _clipboardRepository.GetAllByUserId(userId.ToString());
        }
    }
}